src/main/java/teammates/common/datatransfer/AccountAttributes.java

```diff
@@ -65,6 +65,22 @@ public AccountAttributes(String googleId, String name, boolean isInstructor,
         this.studentProfile.googleId = this.googleId;
     }
     
+    /**
+     * Gets a deep copy of this object.
+     */
+    public AccountAttributes getCopy() {
+        boolean isStudentProfileNull = this.studentProfile == null;
+        if (isStudentProfileNull) {
+            this.studentProfile = new StudentProfileAttributes();
+        }
+        AccountAttributes copy = new AccountAttributes(this.toEntity());
+        if (isStudentProfileNull) {
```

Comment:
Take a look at the definition of `toEntity()`, which I don't intend to modify now:
```java
@Override
public Account toEntity() {
    Assumption.assertNotNull(this.studentProfile); // this line!
    return new Account(googleId, name, isInstructor, email, institute, (StudentProfile) studentProfile.toEntity());
}
```